src/aro/Parser.zig

```diff
@@ -3248,15 +3325,19 @@ fn initializer(p: *Parser, init_ty: Type) Error!Result {
         return error.ParsingFailed;
     }
 
-    var il: InitList = .{};
-    defer il.deinit(p.gpa);
+    if (init_ty.isComplex()) {
+        return p.complexInitializer(init_ty);
```

Comment:
Shouldn't this be handled somewhere in `initializerItem` instead? How are complex numbers in structs handled?

Reply:
Those still work. I don't love that complex initializers are on a separate path but the problem I ran into is that braced initializers (`Parser.initializer`) do not return a Value in the result, because they use initializerItem / convertInitList. I'm not super familiar with the initializer code but it seemed like it would be tricky to add that. If you think it wouldn't be, I'm happy to take another look at it.